File: internal/pkg/dto/request/automation.go
package request

type AutomationCondition struct {
	Field string `json:"field"`
	Value string `json:"value"`
}
type AutomationAction struct {
	Type    string `json:"type"`
	Value   string `json:"value"`
	OwnerID int64  `json:"ownerId"`
}
type AutomationDefinition struct {
	Trigger             string                `json:"trigger"`
	Match               string                `json:"match"`
	OncePerConversation bool                  `json:"oncePerConversation"`
	Conditions          []AutomationCondition `json:"conditions"`
	Actions             []AutomationAction    `json:"actions"`
}
type SaveAutomation struct {
	ID         int64                `json:"id"`
	Revision   int64                `json:"revision"`
	Name       string               `json:"name"`
	Priority   int                  `json:"priority"`
	Definition AutomationDefinition `json:"definition"`
}
type ChangeAutomation struct {
	ID       int64 `json:"id"`
	Revision int64 `json:"revision"`
	Enabled  bool  `json:"enabled"`
}
type TestAutomation struct {
	Definition AutomationDefinition `json:"definition"`
	Channel    string               `json:"channel"`
	Text       string               `json:"text"`
	LeadTags   []string             `json:"leadTags"`
}

File: web/lib/api/automation.ts
import { request } from "@/lib/api/client";

export type AutomationCondition = { field: string; value: string };
export type AutomationAction = { type: string; value: string; ownerId: number };
export type AutomationDefinition = {
  trigger: string;
  match: string;
  oncePerConversation: boolean;
  conditions: AutomationCondition[];
  actions: AutomationAction[];
};
export type AutomationRule = { id: number; revision: number; name: string; priority: number; enabled: boolean; definition: AutomationDefinition; updatedAt: string };
export type AutomationDraft = Pick<AutomationRule, "id" | "revision" | "name" | "priority" | "definition">;
export type AutomationRun = { id: number; ruleId: number; ruleName: string; revision: number; conversationId: number; eventId: number; status: string; results: string[] | null; errorCode: string; attempts: number; createdAt: string; updatedAt: string; definition: AutomationDefinition };
const base = "/api/dashboard/automation";
const post = <T,>(path: string, body: unknown) => request<T>(base + path, { method: "POST", body: JSON.stringify(body) });
export const fetchAutomationRules = () => request<AutomationRule[]>(base + "/list");
export const saveAutomation = (draft: AutomationDraft) => post<AutomationRule>("/save", draft);
export const changeAutomation = (rule: AutomationRule, enabled: boolean) => post<void>("/change", { id: rule.id, revision: rule.revision, enabled });
export const deleteAutomation = (rule: AutomationRule) => post<void>("/delete", { id: rule.id, revision: rule.revision });
export const retryAutomation = (id: number) => post<void>("/retry", { id });
export const fetchAutomationRuns = (ruleId: number, page: number, limit: number) => request<{ results: AutomationRun[]; page: { page: number; limit: number; total: number } }>(`${base}/runs?ruleId=${ruleId}&page=${page}&limit=${limit}`);
export const testAutomation = (definition: AutomationDefinition, channel: string, text: string, leadTags: string[]) => post<{ matched: boolean; conditions: boolean[] }>("/test", { definition, channel, text, leadTags });
